### PR TITLE
doc3721: expose LH resource reservation to harvester-configuration

### PR DESCRIPTION
**Pull request by w13915984028**
expose LH resource reservation to harvester-configuration
expose default storage class replica count to harvester-configuration

issue:
https://github.com/harvester/harvester/issues/3721

source code PR:
Original PR ( https://github.com/harvester/harvester-installer/pull/466 ) is merged into https://github.com/harvester/harvester-installer/pull/402

epic:
[harvester/harvester#3262](https://github.com/harvester/harvester/issues/3262)

### CODE REVIEW

**Review by vickyhella — COMMENTED** [review]
Just some small suggestions

File: docs/install/harvester-configuration.md [hunk 1]
```diff
@@ -515,6 +515,87 @@ install:
...
+
+#### Definition
+
+Sets the default storage class replica count.
+
+Default: 12
+
+Supported values: 0~12. All other values are reverted to 12.
+
+This integer value indicates what percentage of the total allocatable CPU on each node will be reserved for each engine manager Pod.
```
[the remark this answers is not in the document]
Comment (w13915984028) [reply docs/install/harvester-configuration.md:555]: This line is copied from LH doc, prefer to keep unchanged.

https://longhorn.io/docs/1.4.1/references/settings/#guaranteed-engine-manager-cpu
image

File: docs/install/harvester-configuration.md [hunk 2]
```diff
@@ -515,6 +515,87 @@ install:
...
+
+#### Definition
+
+Sets the default storage class replica count.
+
+Default: 12
+
+Supported values: 0~12. All other values are reverted to 12.
+
+This integer value indicates what percentage of the total allocatable CPU on each node will be reserved for each replica manager Pod.
```
[the remark this answers is not in the document]
Comment (w13915984028) [reply docs/install/harvester-configuration.md:583]: similar, copied from LH doc, prefer to keep unchanged.

**Review by vickyhella — APPROVED** [review]

**Review by guangbochen — APPROVED** [review]
LGTM, thanks.

File: docs/install/harvester-configuration.md [hunk 3]
```diff
@@ -515,6 +515,87 @@ install:
   data_disk: /dev/sdb
 ```
 
+### `install.harvester.storageClass.replicaCount`
```
Comment (guangbochen) [inline docs/install/harvester-configuration.md:518]: please use snake_case instead.

File: docs/install/harvester-configuration.md [hunk 4]
```diff
@@ -515,6 +515,87 @@ install:
...
+
+Please refer to [longhorn-replica-count](https://longhorn.io/docs/1.4.1/references/settings/#default-replica-count) for more details.
+
+#### Example
+
+```yaml
+install:
+  harvester:
+    storageClass:
+      replicaCount: 1
```
Comment (guangbochen) [inline docs/install/harvester-configuration.md:540]:
```suggestion
    storage_class:
      replica_count: 1
```

File: docs/install/harvester-configuration.md [hunk 5]
```diff
@@ -515,6 +515,87 @@ install:
...
+```
+
+### `install.harvester.longhorn.defaultSettings.guaranteedEngineManagerCPU`
```
Comment (guangbochen) [inline docs/install/harvester-configuration.md:543]: ditto

File: docs/install/harvester-configuration.md [hunk 6]
```diff
@@ -515,6 +515,87 @@ install:
...
+Before setting the value, please refer to [longhorn-guaranteed-engine-manager-cpu](https://longhorn.io/docs/1.4.1/references/settings/#guaranteed-engine-manager-cpu) for more details.
+
+#### Example
+
+```yaml
+install:
+  harvester:
+    longhorn:
+      defaultSettings:
+        guaranteedEngineManagerCPU: 6
```
Comment (guangbochen) [inline docs/install/harvester-configuration.md:568]: ditto

File: docs/install/harvester-configuration.md [hunk 7]
```diff
@@ -515,6 +515,87 @@ install:
...
+Before setting the value, please refer to [longhorn-guaranteed-replica-manager-cpu](https://longhorn.io/docs/1.4.1/references/settings/#guaranteed-replica-manager-cpu) for more details.
+
+#### Example
+
+```yaml
+install:
+  harvester:
+    longhorn:
+      defaultSettings:
+        guaranteedReplicaManagerCPU: 6
```
Comment (guangbochen) [inline docs/install/harvester-configuration.md:596]: ditto

File: docs/install/harvester-configuration.md (hunk 3 above)
Comment (guangbochen) [reply docs/install/harvester-configuration.md:518]: this is an inherited design from k3os(harvester's original baseOS), and making it consistent with the existing doc format.

**Comment by guangbochen** [comment]
verified new cluster install with the following config, and it works as expected with those new config:
```
scheme_version: 1
install:
  data_disk: /dev/disk/by-id/scsi-0HP_LOGICAL_VOLUME_01000000
  harvester:
    storage_class:
      replica_count: 1
    longhorn:
      default_settings:
        guaranteedReplicaManagerCPU: 6
        guaranteedEngineManagerCPU: 6
```
image
image

**Review by bk201 — APPROVED** [review]
LGTM, thanks.

**Comment by guangbochen** [comment]
@w13915984028 we will merge this PR and feel free to leave a comment if u see any problems, thanks.

**Pull request merged**